src/NexusMods.App.UI/Extensions/R3Extensions.cs

```diff
@@ -8,6 +9,37 @@ namespace NexusMods.App.UI.Extensions;
 [PublicAPI]
 public static class R3Extensions
 {
+    [MustDisposeResource] public static IDisposable WhenActivated<T>(
+        this T obj,
+        Action<T, CompositeDisposable> block)
+        where T : ReactiveR3Object
```

Comment:
Just for curiosity, what is the reason to keep this separate from `ReactiveR3Object`?

Reply:
What do you mean with "keep it separate"?

Reply:
Why is this an extension method rather than being defined directly on ReactiveR3Object?

Reply:
`this` is only available in static classes. Making this method an extension method means you can use `this.WhenActivated`. If I put this method inside `ReactiveR3Object` you have to use `WhenActivated(this)`. Slightly different usage, but I think `this.WhenActivated` is easier to adapt.

Reply:
Ah, I see, makes sense. Thanks for explaining, I appreciate it!